github/repos.go

```diff
@@ -728,6 +728,9 @@ type Protection struct {
 	RequiredPullRequestReviews *PullRequestReviewsEnforcement `json:"required_pull_request_reviews"`
 	EnforceAdmins              *AdminEnforcement              `json:"enforce_admins"`
 	Restrictions               *BranchRestrictions            `json:"restrictions"`
+	RequiredLinearHistory      bool                           `json:"required_linear_history"`
```

Comment:
Each of these three `bool` fields need to be structures with a single `Enabled` field, based on the GitHub v3 API developer docs found here:
https://developer.github.com/v3/repos/branches/#get-branch-protection

Here is the example output from this endpoint:

```
...
  "required_linear_history": {
    "enabled": true
  },
  "allow_force_pushes": {
    "enabled": true
  },
  "allow_deletions": {
    "enabled": true
  }
```

Since each of these structs appear to only have a single `Enabled` field, I'm fine with making them all the same struct type... something like:
```
type EnabledType struct {
  Enabled bool `json:"enabled"`
}
```

and then these three fields would have struct pointers with `omitempty`, like this:

```
RequiredLinearHistory *EnabledType `json:"required_linear_history,omitempty"`
AllowForcePushes      *EnabledType `json:"allow_force_pushes,omitempty"`
AllowDeletions        *EnabledType `json:"allow_deletions,omitempty"`
```

That way, if GitHub provides the values, great... but if not, we know that they weren't returned.